src/Kit/System/Watchdog/IWatchedEventLoop.h

```diff
@@ -0,0 +1,190 @@
+#ifndef KIT_SYSTEM_WATCHDOG_IWATCHEDEVENTLOOP_H_
+#define KIT_SYSTEM_WATCHDOG_IWATCHEDEVENTLOOP_H_
+/*------------------------------------------------------------------------------
+ * Copyright Integer Fox Authors
+ *
+ * Distributed under the BSD 3 Clause License. See the license agreement at:
+ * https://github.com/Integerfox/kit.core/blob/main/LICENSE
+ *
+ * Redistributions of the source code must retain the above copyright notice.
+ *----------------------------------------------------------------------------*/
+/** @file */
+
+#include "kit_config.h"
+
+///
+namespace Kit {
+///
+namespace System {
+
+// Forward declaration - EventLoop is in Kit::System namespace
+class EventLoop;
```

Comment:
`EventLoop` is a child of `TimerManager` - The interface only needs a `TimerManager` for its 'health timer'
```suggestion
class TimerManager;
```